playbooks/provision_dbserver.yml

```diff
@@ -0,0 +1,93 @@
+---
+
+- name: fetch secrets
+  hosts: ofn_servers
+  remote_user: "{{ user }}"
+
+  tasks:
+    - name: update local secrets files
+      include_role:
+        name: fetch_secrets
+        apply:
+          delegate_to: 127.0.0.1
+      when: ( fetch_ofn_secrets is defined or lookup('env','FETCH_OFN_SECRETS') == 'TRUE' ) and inventory_hostname not in groups['local']
+      run_once: true
+
+- name: provision dbserver
+  hosts: db-servers
+  remote_user: "{{ user }}"
+
+  handlers:
+    - include: ../roles/shared_handlers/handlers/main.yml
+
+  roles:
+    - role: ssh_keys
+      become: yes
+      tags: ssh_keys
+
+    - role: geerlingguy.security
+      become: yes
+      become_user: root
+      tags: security
+
+    - role: language
+      tags: language
+
+  tasks:
+    - name: provision server keys
+      copy:
+        src: "{{ inventory_dir + '/../files/dbserver_keys/' + dbserver_key }}"
+        dest: "{{ postgresql_data_dir }}/{{ dbserver_key }}"
+        owner: postgres
+        group: postgres
+        mode: 0600
+      become: yes
+      become_user: root
+      loop:
+        - root.crt
+        - server.crt
+        - server.key
+      loop_control:
+        loop_var: dbserver_key
+      notify: restart local postgres
+
+    - name: configure wal2json
+      include_role:
+        name: dbserver
+        tasks_from: wal2json.yml
+
+    - name: configure postgres
+      import_role:
+        name: geerlingguy.postgresql
+      become: yes
+      vars:
+        postgresql_databases:
+          - name: "{{ db }}"
+            owner: "{{ db_user }}"
+            login_password: "{{ db_password }}"
+            lc_collate: "{{ postgres_encoding }}"
+            lc_ctype: "{{ postgres_encoding }}"
+        postgresql_users:
+          - name: "{{ db_user }}"
+            password: "{{ db_password }}"
+            role_attr_flags: "{{ db_user_roles }}"
+      tags: postgres
+
+    - name: ensure postgres conf.d directory exists
+      file:
+        dest: "{{ postgresql_config_path }}/conf.d"
+        state: directory
+        owner: postgres
+        group: postgres
+        mode: 0740
+      become: yes
+
+    - meta: flush_handlers # Ensure handlers run successfully before reporting success
+
+    - name: notify slack
+      slack:
+        token: "T02G54U79/BF25P9F7A/DJdtYaLLUpRJPiu72d8NqgGg"
+        msg: 'DB server {{ inventory_hostname }} provisioned'
+        channel: "#devops-notifications"
+        username: "ansible executed by {{ lookup('env','USER') }}"
+      when: inventory_hostname not in groups['local']
```

Comment:
why not all this in provision.yml?

Reply:
Some of it is also in `provision.yml` (applying new config changes etc), but we have to set up and provision a totally different server here, and we probably want to be able to do that independently of the app server, right?